orientdb/src/main/scala/akka/stream/alpakka/orientdb/OrientDBSourceStage.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.orientdb
+
+import akka.stream.{Attributes, Outlet, SourceShape}
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+import com.orientechnologies.orient.`object`.db.OObjectDatabaseTx
+import com.orientechnologies.orient.core.command.OCommandResultListener
+import com.orientechnologies.orient.core.sql.query.{OSQLNonBlockingQuery, OSQLSynchQuery}
+import scala.collection.JavaConverters._
+
+final case class OutgoingMessage[T](oDocument: T)
+
+case class OSQLResponse[T](error: Option[String], result: Option[OSQLResult[T]])
+case class OSQLResult[T](records: Seq[OutgoingMessage[T]])
+
+trait MessageReader[T] {
+  def convert(oDocs: List[T]): OSQLResponse[T]
+}
+
+private[orientdb] final class OrientDBSourceStage[T](className: String,
+                                                     query: String,
+                                                     settings: OrientDBSourceSettings,
+                                                     reader: MessageReader[T],
+                                                     clazz: Option[Class[T]] = None)
+    extends GraphStage[SourceShape[OutgoingMessage[T]]] {
+
+  val out: Outlet[OutgoingMessage[T]] = Outlet("OrientDBSource.out")
+  override val shape = SourceShape(out)
+
+  override def createLogic(inheritedAttributes: Attributes) =
+    new OrientDBSourceLogic[T](className,
+                               query,
+                               settings,
+                               out,
+                               shape,
+                               reader,
+                               SkipAndLimit(settings.skip, settings.limit),
```

Comment:
Why using this wrapper object instead of simply reading this properties from the settings?